Add markdown feature

diego-rapoport opened the pull request:
<!--

## Checklist

Please make all of these are true before submitting a pull request.

- My PR branch was based on the `dev` branch
- I am submitting this PR against the `dev` branch
- I wrote or updated any tests related to my changes
- I ran `npm run cq` to format and test my code

-->

## Type of Pull Request

- [ ] Bug fix
- [x] Enhancement

Related Issue #s or links (if any): 

## Description of Changes
Just added a Markdown Previewer to the menu. Fully functional. If more tests are needed, please let me know.
Closes #10

dlford reviewed (changes requested):
@diego-rapoport Great job on this! I think we just need to have the layout stack on mobile, it gets too tight having it side by side.

diego-rapoport commented:
> @diego-rapoport Great job on this! I think we just need to have the layout stack on mobile, it gets too tight having it side by side.

Thank you very much! I'm so glad I could contribute.
 I'll make those changes right away

diego-rapoport commented:
@dlford How about that?

![Recording 2022-10-08 at 22 37 42](https://user-images.githubusercontent.com/66575787/194733548-6ea3b304-ea59-4566-a707-8232b8ef4f46.gif)

I had to make a global css on this, but it's really simple and straight forward.

dlford commented:
@diego-rapoport I think this would be cleaner without the CSS file, you can accomplish this with the `sx` prop on MUI components (https://mui.com/system/getting-started/the-sx-prop/).

Also, looks like there is a linting error in the latest deploy log.

diego-rapoport commented:
Oh, sorry. I thought I had done the push to my fork only. My mistake.

dlford commented:
> Oh, sorry. I thought I had done the push to my fork only. My mistake.

No worries, you did, Netlify deploys previews of PR code automatically.

dlford reviewed (changes requested):
@diego-rapoport this looks good, can you please do the following so we can get it merged:
1. Run `npm install` (looks like `package-lock.json` is out of sync, this will fix it)
2. Run `npm run cq` and address any issues
3. Commit and push those changes

dlford commented:
@diego-rapoport not sure if you saw my last comment? We just need to fix those couple things and then I can merge this PR.

diego-rapoport commented:
> @diego-rapoport not sure if you saw my last comment? We just need to fix those couple things and then I can merge this PR.

Hey! I'm truly sorry. Just got really busy at work and still couldn't make it. I hope by the end of the week I could focus on that.

dlford commented:
@diego-rapoport no worries, thanks!

diego-rapoport commented:
I really thought that I had the Lint and Test phase right. If I run locally my fork(already did the pull and updated it) `npm ci` it doesn't give me any errors.
Can you help me find what's wrong @dlford? I already updated the `package-lock.json` file and it seamed to be good.

dlford commented on __TESTS__/markdown.spec.tsx:
```diff
@@ -0,0 +1,30 @@
+
+test('previews markdown correctly', async () => {
+  render(<MarkDownPreview />);
+  const input = screen.getByRole('textbox');
```
This test is failing because there are two text boxes on the page, let's use `getByLabelText` here instead.

dlford reviewed (changes requested):
@diego-rapoport Just one minor issue with the test, then we are good to merge, thanks!

dlford commented:
@diego-rapoport sorry to nit-pick here, but can we also give that input a label for accessibility?

diego-rapoport commented:
> @diego-rapoport sorry to nit-pick here, but can we also give that input a label for accessibility?

Sure! 
What label should I give? How should I place it as well?
If you can give me more details about how and what should I put that would be great.

dlford commented on pages/markdown.tsx:
```diff
@@ -0,0 +1,55 @@
+          flexDirection: { md: 'row', xs: 'column' },
+        }}
+      >
+        {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
```
delete this line to re-enable the lint rule

dlford commented on pages/markdown.tsx:
```diff
@@ -0,0 +1,55 @@
+        }}
+      >
+        {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
+        <label htmlFor='inputText' />
```
Delete this empty label tag

dlford commented on pages/markdown.tsx:
```diff
@@ -0,0 +1,55 @@
+        {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
+        <label htmlFor='inputText' />
+        <TextField
+          id='inputText'
```
Add a `label` attribute, like the examples here - https://mui.com/material-ui/react-text-field/

dlford replied:
Let's make that a meaningful label, like "Markdown"?

diego-rapoport replied:
That's why I asked **what** to write... but ok.
